src/main/cpp/src/get_json_object.hpp

```diff
@@ -0,0 +1,1210 @@
+  //     p.next_token();
+  //     // JSON validation check
+  //     if (json_token::ERROR == p.get_current_token()) { return false; }
+
+  //     return path_evaluator::evaluate_path(p, g, style, path_ptr + 1, path_size - 1);
+  //   }
+  //   // case _ =>
+  //   // case path 12
+  //   else {
+  //     if (!p.try_skip_children()) { return false; }
+  //     return false;
+  //   }
+  // }
+
+  /**
+   *
+   * This function is rewritten from above commented recursive function.
+   * this function is equivalent to the above commented recursive function.
+   */
+  static CUDF_HOST_DEVICE bool evaluate_path(json_parser<>& p,
+                                             json_generator<>& root_g,
+                                             write_style root_style,
+                                             path_instruction const* root_path_ptr,
+                                             int root_path_size)
+  {
+    // manually maintained context stack in lieu of calling evaluate_path recursively.
+    struct context {
+      // current token
+      json_token token;
+
+      // which case path that this task is from
+      int case_path;
+
+      // used to save current generator
+      json_generator<> g;
+
+      write_style style;
+      path_instruction const* path_ptr;
+      int path_size;
+
+      // is this context task is done
+      bool task_is_done = false;
+
+      // whether written output
+      // if dirty > 0, indicates success
+      int dirty = 0;
+
+      // for some case paths
+      bool is_first_enter = true;
+
+      // used to save child JSON generator for case path 8
+      json_generator<> child_g;
+
+      CUDF_HOST_DEVICE context()
+        : token(json_token::INIT),
+          case_path(-1),
+          g(json_generator<>()),
+          style(write_style::raw_style),
+          path_ptr(nullptr),
+          path_size(0)
+      {
+      }
+
+      CUDF_HOST_DEVICE context(json_token _token,
+                               int _case_path,
+                               json_generator<> _g,
+                               write_style _style,
+                               path_instruction const* _path_ptr,
+                               int _path_size)
+        : token(_token),
+          case_path(_case_path),
+          g(_g),
+          style(_style),
+          path_ptr(_path_ptr),
+          path_size(_path_size)
+      {
+      }
+
+      CUDF_HOST_DEVICE context& operator=(const context& other)
+      {
+        token          = other.token;
+        case_path      = other.case_path;
+        g              = other.g;
+        style          = other.style;
+        path_ptr       = other.path_ptr;
+        path_size      = other.path_size;
+        task_is_done   = other.task_is_done;
+        dirty          = other.dirty;
+        is_first_enter = other.is_first_enter;
+        child_g        = other.child_g;
+
+        return *this;
```

Comment:
This seems to have nothing special, so just leave the compiler do it.
```suggestion
      CUDF_HOST_DEVICE context& operator=(context const&) = default;
```

Reply:
Done.